# From https://github.com/oracle/solaris-userland/blob/master/components/rsyslog/patches/

# This patch is a fix for proper uncompression
# source: written internally
# Should be sent upstream, but hasn't been.
# Bug has existed a long time (probably since rsyslog implemented compression),
# last noticed in 8.4.0
diff -wpruN '--exclude=*.orig' a~/runtime/parser.c a/runtime/parser.c
--- a~/runtime/parser.c	1970-01-01 00:00:00
+++ a/runtime/parser.c	1970-01-01 00:00:00
@@ -322,8 +322,10 @@ static rsRetVal uncompressMessage(smsg_t
 
 	/* we first need to check if we have a compressed record. If so,
 	 * we must decompress it.
+	 * If NO_PRI_IN_RAW is set, this is a local Solaris message and
+	 * therefore not compressed (though it may well begin with 'z'!).
 	 */
-	if(lenMsg > 0 && *pszMsg == 'z') { /* compressed data present? (do NOT change order if conditions!) */
+	if(lenMsg > 0 && *pszMsg == 'z' && pMsg->msgFlags & NO_PRI_IN_RAW == 0) { /* compressed data present? (do NOT change order if conditions!) */
 		/* we have compressed data, so let's deflate it. We support a maximum
 		 * message size of iMaxLine. If it is larger, an error message is logged
 		 * and the message is dropped. We do NOT try to decompress larger messages
